### PR TITLE
longhorn-engine refactor

### PR DESCRIPTION
**Pull request by shuo-wu**

### CODE REVIEW

**Comment by yasker** [comment]
The engine manager needs to clean up after the engine even it's gone without cleanup.

File: scripts/package [hunk 1]
```diff
@@ -1,3 +1,4 @@
+
```
Comment (yasker) [inline scripts/package]: Remove the blank line.

File: trash.lock [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import:
```
Comment (yasker) [inline trash.lock]: ?

File: generate_grpc.sh [hunk 3]
```diff
@@ -1,12 +1,17 @@
 #!/bin/bash
 
+mkdir -p $GOPATH/src/github.com/protocolbuffers && git clone https://github.com/protocolbuffers/protobuf.git $GOPATH/src/github.com/protocolbuffers/protobuf
+
 # controller
-protoc -I controller/rpc/pb -I vendor/github.com/golang/protobuf/ptypes/  controller/rpc/pb/controller.proto --go_out=plugins=grpc:controller/rpc/pb
+protoc -I controller/rpc/pb -I $GOPATH/src/github.com/protocolbuffers/protobuf/src/  controller/rpc/pb/controller.proto --go_out=plugins=grpc:controller/rpc/pb
```
Comment (yasker) [inline generate_grpc.sh]: Need to change back to use vendor.

**Comment by yasker** [comment]
LGTM. I've rebased processlauncher branch to the latest

**Comment by yasker** [comment]
`launch-simple-longhorn` doesn't work.

File: package/launch-simple-longhorn [hunk 4]
```diff
@@ -28,11 +28,10 @@ then
         frontend="tgt-blockdev"
 fi
 
-launcher="longhorn controller"
-extra_opts=""
+longhorn-instance-manager deamon &
```
Comment (ttpcodes) [inline package/launch-simple-longhorn]: This is misspelled (it's supposed to be `daemon`, and I suspect it's one of the reasons why `launch-simple-longhorn` doesn't work.

**Pull request merged**